__tests__/CollapsedMetrics.test.js

```diff
@@ -0,0 +1,611 @@
+            reps: [
+                {
+                    isValid: false,
+                    removed: false,
+                }
+            ]
+        }
+
+        let result = sut.getAvgVelocities(set);
+
+        expect(result).toEqual([]);
+    });
+
+    test('return [] when rep is removed', () => {
+        let set = {
+            reps: [
+                {
+                    isValid: true,
+                    removed: true,
+                }
+            ]
+        }
+
+        let result = sut.getAvgVelocities(set);
+
+        expect(result).toEqual([]);
+    });        
+});
+
+describe('getAvgofAvgVelocities', () => {
+
+    test('null when no avg velocities', () => {
+        let set = {
+            reps: [],
+        }
+
+        let result = sut.getAvgofAvgVelocities(set);
+
+        expect(result).toBe(null);
+    });
+
+    test('return avg of avg velocities', () => {
+        let set = {
+            reps: [
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.833368]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 38, 1.821198]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 39, 1.08]
+                },
+            ]
+        }
+
+        let result = sut.getAvgofAvgVelocities(set);
+
+        expect(result).toBe("1.58");
+    });
+});
+
+describe('getAbsLossVelocity', () => {
+
+    test('return null when avgVs empty', () => {
+        let set = {
+            reps: [],
+        };
+        
+        let result = sut.getAbsLossVelocity(set);
+        
+        expect(result).toBe(null);        
+    });
+
+    test('returns abslossvelocity', () => {
+        let set = {
+            reps: [
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.833368]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 38, 1.821198]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 39, 1.08]
+                },
+            ]
+        }
+        
+        let result = sut.getAbsLossVelocity(set);
+
+        expect(result).toBe(0.75);
+    });
+});
+
+describe('getFirstRepAvgVelocity', () => {
+    
+    test('return null when avgVs empty', () => {
+        let set = {
+            reps: [],
+        };
+            
+        let result = sut.getFirstRepAvgVelocity(set);
+            
+        expect(result).toBe(null);        
+    });
+    
+    test('returns first rep avg velocity', () => {
+        let set = {
+            reps: [
+                {
+                    isValid: true,
```

Comment:
- [ ]   modify test to have a invalid and removed values on the "first" and "second" rep that aren't used